tests/integrations/gcp/test_gcp.py

```diff
@@ -0,0 +1,320 @@
+"""
+# GCP Cloud Functions system tests
+
+"""
+import json
+import os
+import time
+from textwrap import dedent
+import requests
+import uuid
+import tempfile
+import shutil
+import sys
+import subprocess
+
+import pytest
+import pickle
+import os.path
+import os
+from googleapiclient.discovery import build
+from google_auth_oauthlib.flow import InstalledAppFlow
+from google.auth.transport.requests import Request
+
+SCOPES = [
+    "https://www.googleapis.com/auth/cloud-platform",
+    "https://www.googleapis.com/auth/cloud-platform.read-only",
+    "https://www.googleapis.com/auth/cloudfunctions",
+    "https://www.googleapis.com/auth/logging.read",
+    "https://www.googleapis.com/auth/logging.admin",
+]
+
+FUNCTIONS_PRELUDE = """
+import sentry_sdk
+from sentry_sdk.integrations.gcp import GcpIntegration
+import json
+import time
+
+from sentry_sdk.transport import HttpTransport
+
+def event_processor(event):
+    # Adding delay which would allow us to capture events.
+    time.sleep(1)
+    return event
+
+class TestTransport(HttpTransport):
+    def _send_event(self, event):
+        event = event_processor(event)
+        # Writing a single string to stdout holds the GIL (seems like) and
+        # therefore cannot be interleaved with other threads. This is why we
+        # explicitly add a newline at the end even though `print` would provide
+        # us one.
+        print("\\nEVENTS: {}\\n".format(json.dumps(event)))
+
+def init_sdk(timeout_warning=False, **extra_init_args):
+    sentry_sdk.init(
+        dsn="https://123abc@example.com/123",
+        transport=TestTransport,
+        integrations=[GcpIntegration()],
+        shutdown_timeout=10,
+        **extra_init_args
+    )
+"""
+
+
+@pytest.fixture
+def authorized_credentials():
+    credentials = None
+
+    # Skipping tests if environment variables not set.
+    if "SENTRY_PYTHON_TEST_GCP_CREDENTIALS_JSON" not in os.environ:
+        pytest.skip("GCP environ vars not set")
+
+    # The file token.pickle stores the user's access and refresh tokens, and is
+    # created automatically when the authorization flow completes for the first
+    # time.
+    with open(
+        os.environ.get("SENTRY_PYTHON_TEST_GCP_CREDENTIALS_JSON"), "rb"
+    ) as creds_file:
+        for line in creds_file.readlines():
+            creds_json = json.loads(line)
+    project_id = creds_json.get("installed", {}).get("project_id")
+    if not project_id:
+        pytest.skip("Credentials json file is not valid")
+
+    if os.path.exists("token.pickle"):
```

Comment:
Can we avoid using pickle? If possible I'd just stick to JSON.

Reply:
The thing is that, the pickle basically is used so that user doesn't have to re-authenticate its credentials from the browser for every test case. 
So, do we want to do that or keep this usage?

Reply:
no I would just replace `pickle.dumps` with `json.dumps` or something like that

Reply:
This .pickle file contains actually the authentication token and not json. Since we are using that authentication token to authenticate our requests, I feel this can't be avoided. If we avoid this .pickle file, we'll have to verify each test case run through the browser.

Reply:
I understand the purpose of the file but I don't understand why we need to use pickle as a serialization format specifically. It seems to me that we could use any other format, like yaml or json.

Reply:
okay, I got your point, I'll try to do it with json serialization.